### PR TITLE
Add a docker-compose example file (#1943)

### PR DESCRIPTION
**Pull request by mikecaat**
Hi, thanks for developing interesting database. 

As a beginner, a docker-compose example file is very useful because it took me a while to find the way to create a neon cluster on multiple servers.

The docker-compose example file is useful for local testing and beginner tutorials. It creates the following containers, and it supports taking backup pages and WAL to S3 compatible storage, MinIO.

  - etcd x 3
  - pageserver x 1
  - safekeeper x 3
  - compute x 1
  - MinIO x 1        # This is Amazon S3 compatible object storage

Please feel free to comments.

### CODE REVIEW

File: docs/docker.md
Comment (SomeoneToIgnore) [inline docs/docker.md:77]: Can we somehow explicitly set those in the compose file? 
If we plan to keep it as a documentation, users might find it simpler to play with the compose file right away, and creds there would help to discover it sooner.

It also seems that the whole new doc could be placed into the compose file, but I might be wrong :slightly_smiling_face:

File: docker-compose/compute/shell/compute.sh
Comment (SomeoneToIgnore) [inline docker-compose/compute/shell/compute.sh:46]: We usually prefer to have newlines in the end of the files, as least seems that a few maintainers have this setting enabled.
If/When we merge this PR, let's have them added everywhere.

File: docker-compose/docker-compose.yml
Comment (SomeoneToIgnore) [inline docker-compose/docker-compose.yml:99]: Those tags are all hardcoded, and not promised to be permanent.
I think, what is expected here is to have a `build` declaration with a few params to create Docker images from it

File: docker-compose/image/compute/Dockerfile
Comment (mikecaat) [reply docker-compose/image/compute/Dockerfile:1]: Oh, thanks. I'll change to use `Dockerfile.compute-node-v15`

File: docs/docker.md
Comment (mikecaat) [reply docs/docker.md:77]: Does the following is not enough? Or is it better to replace environment variables?
```
  miniocreatebuckets:
    image: minio/mc
    environment:
      - MINIO_ROOT_USER=minio
      - MINIO_ROOT_PASSWORD=password
```

File: docker-compose/compute/shell/compute.sh
Comment (mikecaat) [reply docker-compose/compute/shell/compute.sh:46]: OK, I'll fix it.

File: docker-compose/docker-compose.yml
Comment (mikecaat) [reply docker-compose/docker-compose.yml:99]: OK. I'll try it.

File: docker-compose/image/compute/Dockerfile
Comment (SomeoneToIgnore) [reply docker-compose/image/compute/Dockerfile:1]: Just to be explicit: all `image: neondatabase/neon:2221` in docker-compose has to be pointed to the corresponding Docker files and their builds configured, otherwise we're still relying on some hardcoded tag that's not guaranteed to be stored for long.

File: docs/docker.md
Comment (SomeoneToIgnore) [reply docs/docker.md:77]: Oh, apologies, I must have overlooked that part.

File: docker-compose/docker-compose.yml
Comment (mikecaat) [reply docker-compose/docker-compose.yml]: I used the above configuration.

File: docker-compose/image/compute/Dockerfile
Comment (mikecaat) [reply docker-compose/image/compute/Dockerfile]: OK, thanks. I understood. I changed to use `compute-node-v14`  as default image.
But, I found my docker-compose doesn't work with `compute-node-v15` . I'll check the reason.
```
2022-10-24 10:04:38.477 GMT [39] FATAL:  database files are incompatible with server
2022-10-24 10:04:38.477 GMT [39] DETAIL:  The data directory was initialized by PostgreSQL version 14, which is not compatible with this version 15.0.
```

File: docker-compose/compute/shell/compute.sh
Comment (mikecaat) [reply docker-compose/compute/shell/compute.sh]: I fixed.

File: docker-compose/docker-compose.yml
Comment (mikecaat) [reply docker-compose/docker-compose.yml]: I used a environment variable instead to use `build` declaration. Is it enough now?

File: docker-compose/docker-compose.yml
Comment (SomeoneToIgnore) [reply docker-compose/docker-compose.yml]: Great, let's update the docs (if you plan to leave them): https://github.com/neondatabase/neon/pull/2666/files#diff-ae5611658dcd443d7a30575bebcdf15a71b7a0f5abebb18b62d00aa0373cae74R26

File: docker-compose/docker-compose.yml
Comment (SomeoneToIgnore) [reply docker-compose/docker-compose.yml]: Looks nice, might be good to mention TAG in the docs, however obvious that is for us.

**Review by SomeoneToIgnore — COMMENTED** [review]
Thank you.

One last thing I'd love to see in any for is the automated verification of the whole set-up.

There's still quite an amount of hardcode used, which is ok for now per se, but we would need to detect when things change and this hardcode does not work anymore, so that someone can come and fix it.

No worries if it does not work in the CI, I can take care of it, but please let's make some kind of a script or a python test (similar to what we have in the project already) that works for you locally and can be reproduced by me (so that I could work on its CI runs later)

File: docker-compose/docker-compose.yml
Comment (mikecaat) [reply docker-compose/docker-compose.yml]: fixed.

File: docker-compose/image/compute/Dockerfile
Comment (mikecaat) [reply docker-compose/image/compute/Dockerfile]: fixed to specify `pg_version` when to create new timeline.

File: docker-compose/docker-compose.yml
Comment (mikecaat) [reply docker-compose/docker-compose.yml]: OK. I updated the doc.

**Comment by mikecaat** [comment]
Thanks for your kind help.

I fixed your review comments and so on.

* Update docs
* Support proxy argument when to build docker image
* Support v15 of postgres version
* Add a container to check if the compute is ready or not in docker-compose.yml, and add a test script ( /scripts/docker-compose_test.sh ) which fails if the compute does not ready after 60s.

Please let me know if I forgot anything.
Regards,

**Review by SomeoneToIgnore — APPROVED** [review]
Thank you, I've tested the script and it seems to work.
I would have expected some SQL (similar to what you have in the docs) being run in that test script, do you think it's simple to add one in the tests?

Either way, I think this is a good increment for the first time and now we need to think on the whole set-up better, identify things to improve and work on that.
Also, the CI checks for this are vital to get into, that should be the next step.

I've started the CI for the PR and will leave it to hang by the end of the day if somebody else comes to comment it.
If nothing extraordinary happens, I'll merge the PR before the end of tomorrow.

Thank you again for bringing it up.

**Comment by mikecaat** [comment]
> I would have expected some SQL (similar to what you have in the docs) being run in that test script, do you think it's simple to add one in the tests?

Yes, I agreed. I made a small patch to do so.
Please check if the script works in your environment.

> Either way, I think this is a good increment for the first time and now we need to think on the whole set-up better, identify things to improve and work on that.

Yes, I think so. As a beginner, it's useful document for users to create a neon cluster. 
Thanks a lot for your help!

**Pull request merged**
